src/main/java/com/to/ordermanagementservice/repository/impl/InMemoryOrderRepository.java

```diff
@@ -1,22 +1,25 @@
 package com.to.ordermanagementservice.repository.impl;
 
 import com.to.ordermanagementservice.entity.Order;
+import com.to.ordermanagementservice.repository.OrderItemRepository;
 import com.to.ordermanagementservice.repository.OrderRepository;
 import org.springframework.stereotype.Repository;
 
+import java.time.LocalDate;
 import java.time.OffsetDateTime;
 import java.util.List;
 
 @Repository
 public class InMemoryOrderRepository implements OrderRepository {
 
     private final List<Order> orders;
-
+    private final OrderItemRepository orderItemRepository;
```

Comment:
see the comment above regarding entities linking. Please, inject both Repositories in Order Service and do entity->dto conversion there